File: app/src/main/java/ave/mujica/poc/bglog/BackgroundLogTaskRegistry.java
package ave.mujica.poc.bglog;

import java.util.LinkedHashMap;
import java.util.Map;


public final class BackgroundLogTaskRegistry {
    private static final Map<String, BackgroundLogTask> TASKS = new LinkedHashMap<>();

    

    private BackgroundLogTaskRegistry() {
    }

    public static void register(BackgroundLogTask task) {
        TASKS.put(task.getId(), task);
    }

    public static BackgroundLogTask get(String taskId) {
        return TASKS.get(taskId);
    }
}

File: app/src/main/java/ave/mujica/poc/bglog/BackgroundLogTaskRegistry.kt
package ave.mujica.poc.bglog

object BackgroundLogTaskRegistry {
    private val tasks: MutableMap<String, BackgroundLogTask> = LinkedHashMap()

    fun register(task: BackgroundLogTask) {
        tasks[task.getId()] = task
    }

    fun get(taskId: String?): BackgroundLogTask? {
        return tasks[taskId]
    }
}